hmiddleware/connection_closer.go

```diff
@@ -0,0 +1,76 @@
+package hmiddleware
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"sync"
+
+	"github.com/google/uuid"
+	lru "github.com/hashicorp/golang-lru"
+	"github.com/heroku/x/go-kit/metrics"
+)
+
+type connCloseKey int
+
+var connCloseID connCloseKey
+
+// ConnectionClosingContext adds a unique identifier to the context for a newly established connection. This function
+// is meant to be used as an http.Server's ConnContext field, and is required for the ConnectionClosingMiddleware.
+func ConnectionClosingContext(ctx context.Context, c net.Conn) context.Context {
+	return context.WithValue(ctx, connCloseID, uuid.New().String())
+}
+
+// ConnectionClosingMiddleware returns middleware that will add a "Connection: close" responses header after a single
+// connection has produced maxRequests http requests. The cacheSize determines the size of the LRU cache which tracks
+// connection request counts. This middleware requires the http.Server sets ConnectionClosingContext for it's
+// ConnContext field.
+func ConnectionClosingMiddleware(mp metrics.Provider, maxRequests, cacheSize int) (func(http.Handler) http.Handler, error) {
+	var (
+		closed = mp.NewCounter("server.connection.closes.total")
```

Comment:
the `server` is usually scoped by `METRICS_SUFFIX`
https://github.com/heroku/x/blob/96a1bfd8ec81b0ccb4966c6c8610d9e2ec7a1450/cmdutil/service/standard.go#L169

atm this creates `appname.server.server.connection.closes.total`